Added the option to restore recently closed tabs

d2dyno1 opened the pull request:
This PR enables users to restore recently closed tab. Furthermore, the changes in this PR have been made in mind of future improvements - to allow displaying entire history of recently closed tabs. This PR also brings small optimizations, code flexibility and overall code cleanup related to tabs.

Please leave your opinions on this PR.

Resolves #2117

d2dyno1 commented:
I've noticed an issue with it but I'm not entirely sure how to fix it. The problem lies with invoking the key combination, in switch statement there's additional parameter `tabInstance` of type `bool`. It is sometimes false and sometimes true and I can't find cause of that.

@yaichenbaum

d2dyno1 commented on Files/Views/ModernShellPage.xaml.cs:
```diff
@@ -891,6 +891,16 @@ private async void KeyboardAccelerator_Invoked(KeyboardAccelerator sender, Keybo
                     await storageHistoryHelpers.TryRedo();
                     break;
 
+                case (true, true, false, true, VirtualKey.T): // ctrl + shif + t, restore recently closed tab
```
Here

yaichenbaum replied:
What does `tabInstance` represent?

d2dyno1 replied:
Tab instance is true whether `CurrentPageType` is `GenericFileBrowser` or `GenericFileBrowser`

d2dyno1 replied:
Ok, I solved it by replacing the bool with discard `_` symbol

yaichenbaum commented:
@d2dyno1 Should we add the option to reopen a closed tab to the horizontal tab view context menu?

d2dyno1 commented:
@yaichenbaum Yeah, why not?

yaichenbaum commented:
@d2dyno1 I found two issues,
- `Ctrl + shift + t` crashes the app if you try the shortcut before any tabs were closed.
- `Ctrl + shift + t` crashes the app if the most recently closed tab was the new tab page.

d2dyno1 commented:
@yaichenbaum I've fixed the first issue. I couldn't reproduce the second issue. I don't expect it to crash on home page since I've added a null check in `MainPage.AddNewTabByPathAsync()`
